### ISSUE TITLE
Symfony EntityType with custom entity repository

### ISSUE THREAD

**Issue by raziel057** [phpstan/phpstan-doctrine#255]
Hello,

When using Symfony Form [EntityTypes](https://symfony.com/doc/current/reference/forms/types/entity.html#query-builder), I can notice that the following code doesn't trigger any error:

```php
    $builder->add('badges', EntityType::class, array(
        'class' => Badge::class,
        'choice_label' => 'translations['.$this->locale.'].name',
        'query_builder' => function(EntityRepository $er) {
            return $er->findAll();
        }
));
```
But If I replace ``findAll`` by a custom method ``findAllSortedByName`` coming from my BadgeRepository I got the following error:
```
Call to an undefined method Doctrine\ORM\EntityRepository::findAllSortedByName()
```

Is it supposed or planned to be supported? It would be great as it's commonly used in Symfony forms.

**Comment by VincentLanglet** [comment]
The error is not a bug since you say `$er` is an EntityRepository.

You need to change the type hint of your method
```
function(YourCustomRepository $er) {
     return $er->findAll();
}
```

**Comment by raziel057** [comment]
@VincentLanglet Ah yes seems logical indeed. Thanks for your quick answer.
